lib/defaults.js

```diff
@@ -16,13 +16,14 @@ function setContentTypeIfUnset(headers, value) {
 
 function getDefaultAdapter() {
   var adapter;
-  if (typeof XMLHttpRequest !== 'undefined') {
-    // For browsers use XHR adapter
-    adapter = require('./adapters/xhr');
-  } else if (typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]') {
-    // For node use HTTP adapter
-    adapter = require('./adapters/http');
-  }
+  // For browsers use XHR adapter
+  var xhr = typeof XMLHttpRequest !== 'undefined';
+  // For node use HTTP adapter
+  var http = typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]';
+
+  if (xhr) adapter = require('./adapters/xhr');
+  if (http) adapter = require('./adapters/http');
```

Comment:
Just wondering, you sure we don't need an `else if` here?

```js
if (xhr) {
  adapter = require('./adapters/xhr');
} else if (http) {
  adapter = require('./adapters/http');
}
```

Reply:
I'm sure , We don't need an `else if `
There will be no XMLHttpRequest in Node and no HTTP in the browser

Reply:
Do you have a way for Node to get browsers that have both XMLHttpRequest and HTTP ? 
I don't think that's going to happen